src/json/invoc_result.rs

```diff
@@ -0,0 +1,154 @@
+// Copyright 2019-2023 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use crate::interpreter::{InvocResult, MessageGasCost};
+
+pub mod json {
+    use crate::shim::executor::{Receipt, Trace};
+    use crate::shim::message::Message;
+    use cid::Cid;
+
+    use serde::{Deserialize, Deserializer, Serialize, Serializer};
+
+    use super::*;
+
+    /// Wrapper for serializing and de-serializing an `InvocResult` from JSON.
+    #[derive(Deserialize, Serialize, Debug)]
+    #[serde(transparent)]
+    pub struct InvocResultJson(#[serde(with = "self")] pub InvocResult);
+
+    /// Wrapper for serializing a `InvocResult` reference to JSON.
+    #[derive(Serialize)]
+    #[serde(transparent)]
+    pub struct InvocResultRef<'a>(#[serde(with = "self")] pub &'a InvocResult);
+
+    impl From<InvocResultJson> for InvocResult {
+        fn from(wrapper: InvocResultJson) -> Self {
+            wrapper.0
+        }
+    }
+
+    impl From<InvocResult> for InvocResultJson {
+        fn from(ir: InvocResult) -> Self {
+            InvocResultJson(ir)
+        }
+    }
+
+    #[derive(Serialize, Deserialize)]
+    #[serde(rename_all = "PascalCase")]
+    struct JsonHelper {
```

Comment:
We own `InvocResult`. There's no need for this helper; we can directly control how to serializer `InvocResult`.

Reply:
Agree, we can do that for all the `TraceXXX` types as well and remove all boilerplate code (json helpers and quickcheck tests).

Reply:
done